apps/web/pages/api/auth/verify-email.ts

```diff
@@ -49,9 +68,25 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
     });
 
     if (existingUser) {
-      return res.status(401).json({ message: "A User already exists with this email" });
+      return res.status(401).json({ message: USER_ALREADY_EXISTING_MESSAGE });
     }
 
+    // Ensure this email isn't being added by another user as secondary email
+    const existingSecondaryUser = await prisma.secondaryEmail.findUnique({
+      where: {
+        email: userMetadataParsed?.emailChangeWaitingForVerification,
+      },
+      select: {
+        id: true,
+        userId: true,
+      },
+    });
+
+    if (existingSecondaryUser && existingSecondaryUser.userId !== user.id) {
```

Comment:
NIT: I think it would be better if we added `userId: user.id` to the `where` in the query. That way we can just check if `existingSecondaryUser` is not null

Reply:
@joeauyeung The reason its not there is because of 2 reasons
1. We throw error if other user has it added in the line you highlighted (not equal check)
2. In line 106, we swap the secondary email and old email if the current user added it (equal check)
Inorder to have both the checks, i didn't add that. Hope it makes sense